path.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logicalcluster
+
+import (
+	"encoding/json"
+	"path"
+	"regexp"
+	"strings"
+)
+
+var (
+	// Wildcard is the Path indicating a requests that spans many logical clusters.
+	Wildcard = Path{value: "*"}
+)
+
+const (
+	separator = ":"
+)
+
+// Path represents a colon separated list of words describing a path in a logical cluster hierarchy,
+// like a file path in a file-system.
+//
+// For example in the case the logical cluster hierarchy is implemented via workspaces,
```

Comment:
What's a workspace? :)

Reply:
i'll bring back the old comment

Reply:
This is an example logical cluster hierarchy. There is no predefined hierarchy in this package. Hence, if we want to show these paths, we must formulate it as "for instance".

Reply:
the only term will be a `logical cluster`, the rest (Name and Path) will be structured around it. I want to keep it very simple and short.